x-pack/plugins/timelines/server/search_strategy/index_fields/index.ts

```diff
@@ -223,9 +224,10 @@ export const createFieldItem = (
   beatFields: BeatFields,
   indexesAlias: string[],
   index: FieldSpec,
-  indexesAliasIdx: number
+  indexesAliasIdx: number | null
 ): IndexField => {
-  const alias = indexesAlias[indexesAliasIdx];
+  const alias = indexesAliasIdx != null ? [indexesAlias[indexesAliasIdx]] : indexesAlias;
```

Comment:
if null, this is because `dataViewId` got fields for full index pattern instead of one index at a time.